src/praisonai/praisonai/scheduler/executor.py

```
@@ -292,10 +354,120 @@ async def _execute_one(self, job: "ScheduleJob") -> JobResult:
+            return None
+        try:
+            agent.tools = filtered
+        except Exception as e:  # pragma: no cover - defensive
+            logger.warning("RunPolicy could not scope agent tools: %s", e)
+            return None
+
+        def _restore() -> None:
+            try:
+                agent.tools = original
+            except Exception:  # pragma: no cover - defensive
+                pass
```

Ruff (0.15.18)

[error] `try`-`except`-`pass` detected, consider logging the exception

(S110)